routers/web/web.go

```diff
@@ -1177,6 +1177,10 @@ func RegisterRoutes(m *web.Route) {
 		}, func(ctx *context.Context) {
 			ctx.Data["EnableFeed"] = setting.EnableFeed
 		}, repo.MustBeNotEmpty, reqRepoCodeReader, context.RepoRefByType(context.RepoRefTag, true))
+	}, reqSignIn, context.RepoAssignment, context.UnitTypes())
```

Comment:
I do not think it looks good. And you lose the ability of `m.Post("/tags/delete", repo.DeleteTag, reqSignIn` and other handlers.

Reply:
I didn't edit the `m.Post("/tags/delete"` handler, it's here:
https://github.com/go-gitea/gitea/blob/deafc1afffb399580d49e2af937a0e071d669c2d/routers/web/web.go#L1201-L1202

In this PR, I only add a tags list to show all tags when the release setting is disabled. If users want to create or delete a tag, they can't do these things on the UI and they need to use `git` command to push the changes manually.

Reply:
But users could delete a tag by Web UI

![image](https://user-images.githubusercontent.com/2114189/233828655-8b45c720-5e5a-4b71-9d9b-170ea37951d9.png)

Reply:
Maybe the question is that should we keep the delete button when users cannot create tags on the Web UI. IMO, users may find it strange if they can delete tags but cannot create tags, so I removed the delete button. If anyone thinks we don't need to remove this button, I could add it back.

In fact, I think the better solution is to separate the tag-related logic from the release-related logic so the tag-related functions won't be affected if the release setting is disabled. But it's a big work.